flexget/plugins/clients/tarantool_queue.py

```diff
@@ -0,0 +1,145 @@
+                'additionalProperties': False,
+            },
+        ]
+    }
+
+    def __init__(self):
+        self._queue = None
+        self._is_prepared = False
+        self._config = None
+        self._tarantool_module = None
+
+    @property
+    def config(self):
+        if self._config is None:
+            raise plugin.PluginError('Config is not set', logger)
+        return self._config
+
+    @config.setter
+    def config(self, config):
+        self._config = config
+
+    @property
+    def queue(self):
+        self.prepare()
+        if self._queue is None:
+            self._queue = TarantoolQueue(self._tarantool_module,
+                                         self.config.get('host'),
+                                         self.config.get('port'),
+                                         self.config.get('user'),
+                                         self.config.get('password'))
+
+        return self._queue
+
+    @property
+    def tube(self):
+        return self.queue.tube(self.config.get('tube'))
+
+    def prepare(self):
+        if not self._is_prepared:
+            try:
+                import tarantool
+                self._tarantool_module = tarantool
+            except:
+                raise plugin.PluginError('tarantool_queue module required.', logger)
+
+            if self.config.get('tube') is None:
+                raise plugin.PluginError('Tarantool Queue tube is not defined in config.', logger)
+
+            self._is_prepared = True
+
+class PluginTarantoolQueueInput(PluginTarantoolQueueBase):
+    def on_task_input(self, task, config):
+        if not config.get('enabled', True):
+            return
+
+        self.config = config
+        tarantool_entries = self.tube.take(0)
+        do_continue = True
+        entry_count = 0
+        entries = []
+        while tarantool_entries is not None and len(tarantool_entries) > 0 and do_continue:
+            for tarantool_entry in tarantool_entries:
+                entry_id, entry_type, entry_data = tarantool_entry
+                entry = Entry.deserialize(entry_data, None)
+                entries.append(entry)
+                self.tube.ack(entry_id)
+                entry_count += 1
+                if entry_count > 30:
+                    do_continue = False
+                else:
+                    tarantool_entries = self.tube.take(0)
+
+        return entries
+
+
+class PluginTarantoolQueueOutput(PluginTarantoolQueueBase):
+    def on_task_output(self, task, config):
+        if not config.get('enabled', True):
+            return
+        if not task.accepted:
+            return
+
+        self.config = config
+
+        for entry in task.accepted:
+            try:
+                self.tube.put(Entry.serialize(entry))
+            except Exception as e:
+                entry.fail(e)
+
+@event('plugin.register')
+def register_plugin():
+    plugin.register(PluginTarantoolQueueInput, 'tarantool_queue_input', api_ver=2)
```

Comment:
Plugin naming convention, we don't have `output` or `input` in names. Input should be just `from_tarantool` and output just `tarantool`.